### PR TITLE
Add /makeCall endpoint to support client-to-number/client Dial

### PR DESCRIPTION
**Pull request by bchen-twilio**
Add a new `/placeCall` endpoint to support different functions based on the HTTP param `to`:
- Play a message using `<Say>` if the `to` value is empty.
- Dial to a PSTN number if the first character of `to` value is one of `{+1234567890}`
- Dial to another client otherwise

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -39,21 +41,15 @@ def outgoing():
...
-  account_sid = os.environ.get("ACCOUNT_SID", ACCOUNT_SID)
-  api_key = os.environ.get("API_KEY", API_KEY)
-  api_key_secret = os.environ.get("API_KEY_SECRET", API_KEY_SECRET)
-
-  client = Client(api_key, api_key_secret, account_sid)
-  call = client.calls.create(url=request.url_root + 'incoming', to='client:' + IDENTITY, from_='client:' + CALLER_ID)
-  return str(call.sid)
+    response = twilio.twiml.Response()
+    to = request.form.get('server_param_to', IDENTITY)
+    if to[0] in "+1234567890":
```
Comment (kbagchiGWC) [inline server.py]: Add a comment here. Something like `For simplicity, the following code assumes that identity name starts only with letters and not with numbers.`

File: server.py [hunk 2]
```diff
@@ -39,21 +41,16 @@ def outgoing():
...
-  client = Client(api_key, api_key_secret, account_sid)
-  call = client.calls.create(url=request.url_root + 'incoming', to='client:' + IDENTITY, from_='client:' + CALLER_ID)
-  return str(call.sid)
+    response = twilio.twiml.Response()
+    # For simplicity, the following code assumes that identity name starts only with letters and not with numbers.
+    to = request.form.get('server_param_to', IDENTITY)
+    if to[0] in "+1234567890":
+        response.dial(callerId=CALLER_NUMBER).number(to)
+    else:
+        response.dial(callerId='client:'+CALLER_ID).client(to)
```
Comment (idelgado) [inline server.py]: Rename `CALLER_ID` to `client:quickstart` at the top of the file

File: server.py [hunk 3]
```diff
@@ -39,21 +41,16 @@ def outgoing():
...
 def placeCall():
-  account_sid = os.environ.get("ACCOUNT_SID", ACCOUNT_SID)
-  api_key = os.environ.get("API_KEY", API_KEY)
-  api_key_secret = os.environ.get("API_KEY_SECRET", API_KEY_SECRET)
-
-  client = Client(api_key, api_key_secret, account_sid)
-  call = client.calls.create(url=request.url_root + 'incoming', to='client:' + IDENTITY, from_='client:' + CALLER_ID)
-  return str(call.sid)
+    response = twilio.twiml.Response()
+    # For simplicity, the following code assumes that identity name starts only with letters and not with numbers.
```
Comment (idelgado) [inline server.py]: Support 3 options:
- Null or empty string: returns the standard congratulatory message
- Is "+1234567890" : dials as a number
- Is Not "+1234567890": dials as a client

File: server.py [hunk 4]
```diff
@@ -10,6 +10,8 @@
 PUSH_CREDENTIAL_SID = 'CR***'
 APP_SID = 'AP***'
 
+# Replace the value of `CALLER_NUMBER` with a valid Twilio number under the account.
```
Comment (idelgado) [inline server.py]:
```Python
"""
Use a valid Twilio number by adding to your account via https://www.twilio.com/console/phone-numbers/verified
"""
```
CALLER_NUMBER = "0123456789"

File: server.py [hunk 5]
```diff
@@ -10,6 +10,8 @@
...
+CALLER_NUMBER = '14151234567'
 IDENTITY = 'voice_test'
 CALLER_ID = 'quick_start'
```
Comment (idelgado) [inline server.py]:
```Python
"""
The caller id used when a client is dialed.
"""
```

File: server.py [hunk 6]
```diff
@@ -39,21 +41,16 @@ def outgoing():
...
-  account_sid = os.environ.get("ACCOUNT_SID", ACCOUNT_SID)
-  api_key = os.environ.get("API_KEY", API_KEY)
-  api_key_secret = os.environ.get("API_KEY_SECRET", API_KEY_SECRET)
-
-  client = Client(api_key, api_key_secret, account_sid)
-  call = client.calls.create(url=request.url_root + 'incoming', to='client:' + IDENTITY, from_='client:' + CALLER_ID)
-  return str(call.sid)
+    response = twilio.twiml.Response()
+    # For simplicity, the following code assumes that identity name starts only with letters and not with numbers.
+    to = request.form.get('server_param_to', IDENTITY)
```
Comment (idelgado) [inline server.py]: Rename `server_param_to` to `to`

File: server.py [hunk 7]
```diff
@@ -36,24 +45,20 @@ def token():
...
-
-@app.route('/incoming', methods=['GET', 'POST'])
-def incoming():
-  resp = twilio.twiml.Response()
-  resp.say("Congratulations! You have received your first inbound call! Good bye.")
+  resp.say("Congratulations! You have just made your first oubound call! Good bye.")
   return str(resp)
 
 @app.route('/placeCall', methods=['GET', 'POST'])
 def placeCall():
```
Comment (idelgado) [inline server.py]: `def placeIncomingCall()` <-- place an incoming call

`def makeOutgoingCall()` <-- make an outgoing cal

File: server.py [hunk 8]
```diff
@@ -10,8 +10,17 @@
 PUSH_CREDENTIAL_SID = 'CR***'
 APP_SID = 'AP***'
 
-IDENTITY = 'voice_test'
-CALLER_ID = 'quick_start'
+"""
+Use a valid Twilio number by adding to your account via https://www.twilio.com/console/phone-numbers/verified
+"""
+CALLER_NUMBER = '0123456789'
```
Comment (kbagchiGWC) [inline server.py]: how about making this `1234567890`?

File: server.py [hunk 9]
```diff
@@ -46,15 +46,6 @@ def token():
...
 
-"""
-Creates an endpoint that plays back a greeting when the outbound call is connected.
-"""
-@app.route('/outgoing', methods=['GET', 'POST'])
-def outgoing():
-  resp = VoiceResponse()
-  resp.say("Congratulations! You have made your first oubound call! Good bye.")
-  return str(resp)
-
```
Comment (kbagchiGWC) [inline server.py]: 👍

**Pull request merged**